validators.go

```diff
@@ -0,0 +1,139 @@
+package imgix
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// validateDomain uses Go's url.Parse and url.Hostname functions to
+// validate the domain. Elsewhere we use a regex to filter invalid
+// domains. However, the same regex won't work in this case as Go
+// does not support positive look-a-heads (i.e. `(?=)`).
+// TODO: Discuss. Go doesn't support positive look-a-heads so our
+// domain regex won't work here. I will explore regex alternatives.
+func validateDomain(domain string) (string, error) {
+	if strings.HasPrefix(domain, "http") {
+		u, err := url.Parse(domain)
+		if err != nil {
+			return "", err
```

Comment:
[Go 1.13 introduced error wrapping](https://blog.golang.org/go1.13-errors), which you could use in the validation functions (or elsewhere) to provide more context for the error. Like:

```go
u, err := url.Parse(domain)
if err != nil {
    return "", fmt.Errorf("domain %s failed validation: %w", domain, err)
}
```

If the user calls the `Unwrap()` method of the returned error, they can still access the original error.